uc-crux-llvm/src/UCCrux/LLVM/FullType/Memory.hs

```diff
@@ -52,21 +53,24 @@ import           UCCrux.LLVM.FullType.Type (FullTypeRepr, ModuleTypes, dataLayou
 sizeInBytes ::
   ModuleTypes m ->
   FullTypeRepr m ft ->
-  Integer
+  Natural
 sizeInBytes mts ftRepr =
   let dl = dataLayout mts
       cdl = crucibleDataLayout dl
-  in bytesToInteger (memTypeSize cdl (toMemType dl ftRepr))
+  in bytesToNatural (memTypeSize cdl (toMemType dl ftRepr))
 
 -- | Size in bytes of an array of a given type with a given length.
 arraySizeInBytes ::
   ModuleTypes m ->
   FullTypeRepr m ft ->
   -- | Array length
-  Integer ->
-  Integer
+  Natural ->
+  Natural
 arraySizeInBytes mts ftRepr size = size * sizeInBytes mts ftRepr
 
+natToInt :: Natural -> Integer
+natToInt = fromIntegral  -- safe
```

Comment:
```suggestion
natToInt = toInteger  -- safe
```

We generally prefer `toInteger` when it fits because it is less polymorphic and always safe. Potential future changes (unlikely here) could cause noisy breakage, which is ideal.

Reply:
Thanks for the tip!